Products/ZenModel/DataRoot.py

```diff
@@ -876,7 +876,20 @@ def _updateEmailNotifications(self, REQUEST):
     def zmanage_editProperties(self, REQUEST=None, redirect=False):
         """Handle our authentication mechanism
         """
+        
         if REQUEST:
+            curuser = self.dmd.ZenUsers.getUser().getId()
+            curpasswd = REQUEST.get('curPasswd')
+            
+            if not self.dmd.ZenUsers.authenticateCredentials(curuser, curpasswd):
+            messaging.IMessageSender(self).sendToBrowser(
+                'Error',
+                'Confirmation password is empty or invalid. Please'+
```

Comment:
You don't need a `+` here.  Python automatically concatenates strings found inside parenthesis.